source/opt/types.h

```diff
@@ -369,7 +370,7 @@ class Struct : public Type {
   bool decoration_empty() const override {
     return decorations_.empty() && element_decorations_.empty();
   }
-  const std::unordered_map<uint32_t, std::vector<std::vector<uint32_t>>>&
+  const std::map<uint32_t, std::vector<std::vector<uint32_t>>>&
```

Comment:
Please document that we need an ordered map here for stable traversals.

Reply:
Added the comment on the declaration of the member variable.